pcsx2/PluginManager.cpp

```diff
@@ -434,6 +439,51 @@ static const LegacyApi_OptMethod s_MethMessOpt_GS[] =
 	{ NULL }
 };
 
+std::map<ConsoleColors, uint32_t> colorTable {
+	{Color_Black        , 0xFF000000},
+	{Color_Green        , 0xFF008000},
+	{Color_Red          , 0xFF000080},
+	{Color_Blue         , 0xFF800000},
+	{Color_Magenta      , 0xFF800080},
+	{Color_Orange       , 0xFF004080},
+	{Color_Gray         , 0xFF404040},
+	{Color_Cyan         , 0xFF808000},
+	{Color_Yellow       , 0xFF008080},
+	{Color_White        , 0xFFFFFFFF},
+	{Color_StrongBlack  , 0xFF000000},
+	{Color_StrongRed    , 0xFF0000FF},
+	{Color_StrongGreen  , 0xFF00FF00},
+	{Color_StrongBlue   , 0xFFFF0000},
+	{Color_StrongMagenta, 0xFFFF00FF},
+	{Color_StrongOrange , 0xFF0080FF},
+	{Color_StrongGray   , 0xFFC0C0C0},
+	{Color_StrongCyan   , 0xFFFFFF00},
+	{Color_StrongYellow , 0xFF00FFFF},
+	{Color_StrongWhite  , 0xFFFFFFFF}
+};
+
+template<typename ... Args>
+void OSD_Log(ConsoleColors color, const std::string& format, Args ... args)
+{
+	uint32_t hex = colorTable[color];
+	if(hex == 0) hex = 0xFF00FFFF;
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-security"
+	size_t size = snprintf( nullptr, 0, format.c_str(), args ... ) + 1; // Extra space for '\0'
+	char buf[size];
```

Comment:
`char buf[size];` Doesn't work for Visual Studio. I replaced it with `std::vector<char> buf(size);`
Which then neeeds `buf `on lines 475 and 477 extended to `buf.data()`.